torch/_inductor/kernel/flex_attention.py

```diff
@@ -2513,6 +2533,10 @@ def flex_attention_backward(*args, **kwargs):
 
     set_head_dim_values(kernel_options, qk_head_dim, v_head_dim, V.graph.sizevars)
 
+    # Get available fp32 precision options. ROCm will return TF32 and FP32 to ensure
+    # that the fastest implementation is chosen.
+    fp32_precision = get_float32_precision()
```

Comment:
this seems weird to me I think we should match exactly what the user specifies.. why would fp32 be faster than tf32?

Reply:
Only MI300/MI325 have real TF32 hardware. On all other chips TF32 support is simulated with FP32.

Reply:
Then I think we should not append more options that will be equivalent and rather only append fp32 on those machines